### PR TITLE
docs: added programs/orgUnits endpoint description [DHIS2-10509]

### PR DESCRIPTION
**Pull request by gnespolino**

### CODE REVIEW

File: src/commonmark/en/content/developer/web-api/metadata.md [hunk 1]
```diff
@@ -2315,6 +2316,43 @@ the following resource.
 </tbody>
 </table>
 
+### Get organisation units by programs
+
+<!--DHIS2-SECTION-ID:webapi_organisation_units_by_programs-->
+
+There's a special ad-hoc endpoint built to retrieve associations between Programs and Organisation Units. This should be
```
Comment (Bekkalizer) [inline src/commonmark/en/content/developer/web-api/metadata.md]:
```suggestion
Purpose-built endpoint to retrieve associations between Programs and Organisation Units. This endpoint is
```

File: src/commonmark/en/content/developer/web-api/metadata.md [hunk 2]
```diff
@@ -2315,6 +2316,43 @@ the following resource.
...
+the preferred way to retrieve programs that are either:
```
Comment (Bekkalizer) [inline src/commonmark/en/content/developer/web-api/metadata.md]:
```suggestion
the preferred way to retrieve program org unit associations.
```

File: src/commonmark/en/content/developer/web-api/metadata.md [hunk 3]
```diff
@@ -2315,6 +2316,43 @@ the following resource.
...
+
+- unassigned to any organisation units
```
Comment (Bekkalizer) [inline src/commonmark/en/content/developer/web-api/metadata.md]:
```suggestion

```

File: src/commonmark/en/content/developer/web-api/metadata.md [hunk 4]
```diff
@@ -2315,6 +2316,43 @@ the following resource.
...
 
+### Get organisation units by programs
+
+<!--DHIS2-SECTION-ID:webapi_organisation_units_by_programs-->
+
+There's a special ad-hoc endpoint built to retrieve associations between Programs and Organisation Units. This should be
+the preferred way to retrieve programs that are either:
+
+- unassigned to any organisation units 
+- assigned explicitly to org units within the scope of logged in user capture scope.
```
Comment (Bekkalizer) [inline src/commonmark/en/content/developer/web-api/metadata.md]:
```suggestion

```
I think these two bullets are redundant, as they cover all the known use cases.

File: src/commonmark/en/content/developer/web-api/metadata.md [hunk 5]
```diff
@@ -2315,6 +2316,43 @@ the following resource.
...
+responses will have the following format:
+
+    {
+        "<programUid>": [
+            "<orgUnitUid_1>",
+            "<orgUnitUid_2>",
+            ...,
+            "<orgUnitUid_n>"
+        ],
+        "<anotherProgramUid>": [
```
Comment (Bekkalizer) [inline src/commonmark/en/content/developer/web-api/metadata.md]:
```suggestion
        "<programUid>": [
```

File: src/commonmark/en/content/developer/web-api/metadata.md [hunk 6]
```diff
@@ -2315,6 +2316,43 @@ the following resource.
...
+- assigned explicitly to org units within the scope of logged in user capture scope.
+
+
+    /api/33/programs/orgUnits?programs=<programUid,anotherProgramUid,...>
+
+responses will have the following format:
+
+    {
+        "<programUid>": [
+            "<orgUnitUid_1>",
```
Comment (Bekkalizer) [inline src/commonmark/en/content/developer/web-api/metadata.md]:
```suggestion
            "<orgUnitUid>",
```

File: src/commonmark/en/content/developer/web-api/metadata.md [hunk 7]
```diff
@@ -2315,6 +2316,43 @@ the following resource.
...
+            "<orgUnitUid_2>",
```
Comment (Bekkalizer) [inline src/commonmark/en/content/developer/web-api/metadata.md]:
```suggestion
            "<orgUnitUid>",
```

File: src/commonmark/en/content/developer/web-api/metadata.md [hunk 8]
```diff
@@ -2315,6 +2316,43 @@ the following resource.
...
+            ...,
+            "<orgUnitUid_n>"
```
Comment (Bekkalizer) [inline src/commonmark/en/content/developer/web-api/metadata.md]:
```suggestion
            "<orgUnitUid>"
```

File: src/commonmark/en/content/developer/web-api/metadata.md [hunk 9]
```diff
@@ -2315,6 +2316,43 @@ the following resource.
...
+        ],
+        "<anotherProgramUid>": [
+            "<orgUnitUid_1>",
```
Comment (Bekkalizer) [inline src/commonmark/en/content/developer/web-api/metadata.md]:
```suggestion
            "<orgUnitUid>",
```

File: src/commonmark/en/content/developer/web-api/metadata.md [hunk 10]
```diff
@@ -2315,6 +2316,43 @@ the following resource.
...
+            "<orgUnitUid_2>",
```
Comment (Bekkalizer) [inline src/commonmark/en/content/developer/web-api/metadata.md]:
```suggestion
            "<orgUnitUid>",
```

File: src/commonmark/en/content/developer/web-api/metadata.md [hunk 11]
```diff
@@ -2315,6 +2316,43 @@ the following resource.
...
+            ...,
+            "<orgUnitUid_n>"
```
Comment (Bekkalizer) [inline src/commonmark/en/content/developer/web-api/metadata.md]:
```suggestion
            "<orgUnitUid>"
```

File: src/commonmark/en/content/developer/web-api/metadata.md [hunk 12]
```diff
@@ -2315,6 +2316,43 @@ the following resource.
...
+        ],
+        "...": [
+            ...,
+            ...
+        ],
+        "<yetAnotherProgramUid>": []
```
Comment (Bekkalizer) [inline src/commonmark/en/content/developer/web-api/metadata.md]:
```suggestion
        "<programUid>": []
```

File: src/commonmark/en/content/developer/web-api/metadata.md [hunk 13]
```diff
@@ -2315,6 +2316,43 @@ the following resource.
...
+    }
+
+programs with no associated uids (hence accessible by all orgUnits) are returned with an empty array [] of orgUnits.
```
Comment (Bekkalizer) [inline src/commonmark/en/content/developer/web-api/metadata.md:2352]: Question not directly relevant to the docs, but to the code - is this the same functionality that we see in the old programs endpoint?

**Review by Bekkalizer — CHANGES_REQUESTED** [review]
Hey!
Added a few changes as suggestions. If you think they are OK it should be easy to apply :)

File: src/commonmark/en/content/developer/web-api/metadata.md (hunk 13 above)
Comment (gnespolino) [reply src/commonmark/en/content/developer/web-api/metadata.md:2352]: when the old endpoint was invoked to retrieve program/org units associations (with restrictToCaptureScope=true), the response was the same, in the meaning that for each program, the two endpoints returns the same orgUnits associations. The shape of the responses is different though.

**Review by vilkg — APPROVED** [review]

**Review by Bekkalizer — APPROVED** [review]

**Pull request merged**